### PR TITLE
PCVL-729-bug fix disappearing parameters

### PR DESCRIPTION
**Pull request by raksharuia**

### CODE REVIEW

File: perceval/utils/algorithms/simplification.py [hunk 1]
```diff
@@ -393,12 +394,14 @@ def _simplify_PS(components, m, display):
                 r0 = invert_permutation(perm_list)[r0]
 
             elif r0 in old_r:
+                # a component other than PS or PERM in the same mode as current PS
                 break
 
-        if not found_PS and (c.get_variables()["phi"] % (2 * np.pi) or display):
+        if not PS_found_n_simplified and (float(phi) % (2 * np.pi) or display):
```
Comment (MarionQuandela) [inline perceval/utils/algorithms/simplification.py:400]: I don't understand what this line does :'(

File: perceval/utils/algorithms/simplification.py [hunk 2]
```diff
@@ -364,26 +364,27 @@ def _simplify_PS(components, m, display):
     [r, c] = components.pop()
```
[the remark this answers is not in the document]
Comment (raksharuia) [reply perceval/utils/algorithms/simplification.py:364]: use of [r, c] for components of a processor is everywhere in perceval where the components of a processor are dealt with. I do not know the historic reason for the name. I would rather change all in the same task (not relevant to this PR) and not change only one instance of it here.

[the remark this answers is not in the document]
Comment (raksharuia) [reply perceval/utils/algorithms/simplification.py:364]: docstring of display is present in the main `simplify` method (defined on L36 of this same script) - it is the one that calls all the private methods. I added typehints to all the private methods in this script that are called by `simplify`.

[the remark this answers is not in the document]
Comment (ericbrts) [reply perceval/utils/algorithms/simplification.py]: This is indeed a historical naming used in lots of places in Perceval. `r` stands for "range", as in "range of modes".

**Pull request merged**